compiler-rt/lib/profile/InstrProfilingBuffer.c

```diff
@@ -93,11 +93,15 @@ uint64_t __llvm_profile_get_counters_size(const char *Begin, const char *End) {
 COMPILER_RT_VISIBILITY
 uint64_t __llvm_profile_get_num_bitmap_bytes(const char *Begin,
                                              const char *End) {
+  if (Begin == NULL && End == NULL)
```

Comment:
How about just make this
```
if (Begin == End)
   return 0;
```
Only one comparison and handles more cases without a subtract, if we have to do comparisons anyway.

Reply:
Makes sense, done!

A little bit more context, I decided to do this when working on a profile-only change for type profiling (https://github.com/llvm/llvm-project/pull/81691/files). If type profiling is off (which is the case initially), the pointer to relevant sections are null.